### PR TITLE
Changed default port

### PR DESCRIPTION
**Pull request by UpTide**
Changed the default port to be in the dynamic range, added a check to see… if requested port was out of range, and added an error message if user enters an out of range port.

### CODE REVIEW

**Comment by UpTide** [comment]
Let me know if you have any questions or if I should change anything please!

**Comment by Mailaender** [comment]
:+1: as according to https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers the previous 1234 is part of the registered section and already used. Plus this gets rid of magic numbers in the source code.

File: OpenRA.Game/Settings.cs [hunk 1]
```diff
@@ -39,7 +39,22 @@ public class ServerSettings
...
-		public int ListenPort = 1234;
+		private int listenPort = 49152;
+		public int ListenPort
+		{
+			get
+			{
+				return this.listenPort;
+			}
+
+			set
```
Comment (pchote) [inline OpenRA.Game/Settings.cs]: Please move the validation logic into `ServerCreationLogic`. These settings classes are automatically synced with files on disk, so shouldn't contain any logic that isn't related to this purpose.

**Comment by UpTide** [comment]
I put the `new IPEndPoint()` inside the `Server` constructor to catch if the port number was too high. (It was unhandled before.)
I also utilized the existing error handler and removed the one I added earlier.

**Comment by UpTide** [comment]
I think I might have pushed the standalone server start scripts too by mistake. I was testing them to make sure they still worked after my changes. Let me know if I need to take them out.

File: OpenRA.Game/Server/Server.cs [hunk 2]
```diff
@@ -121,10 +121,13 @@ public void EndGame()
 				t.GameEnded(this);
 		}
 
-		public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData, bool dedicated)
+		public Server(IPAddress address, int port, ServerSettings settings, ModData modData, bool dedicated)
 		{
 			Log.AddChannel("server", "server.log");
 
+			if (port < 49152 | port > 65534)
+				throw new System.Net.Sockets.SocketException(10022);
```
Comment (pchote) [inline OpenRA.Game/Server/Server.cs]: I'm all for hard-blocking ports outside this range from the in-game UI, but somewhere near 100% of our community hosted dedicated servers will also be caught by this.  Considering that we rely on the good-will of these hosts, forcing this on dedicated servers seems like a really bad idea.

Can we please relax this to a warning in the server log?

File: OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs [hunk 3]
```diff
@@ -180,7 +180,7 @@ void CreateAndJoin()
 			var name = Settings.SanitizedServerName(panel.Get<TextFieldWidget>("SERVER_NAME").Text);
 			int listenPort;
 			if (!Exts.TryParseIntegerInvariant(panel.Get<TextFieldWidget>("LISTEN_PORT").Text, out listenPort))
-				listenPort = 1234;
+				listenPort = Game.Settings.Server.ListenPort;
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs]: We can then add a check and specific "Port must be between 49151 and 65535" error dialog here.

Comment (LipkeGu) [reply OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs]: we could put a warning and implement a method which is using 1234 as alternate port

**Comment by reaperrr** [comment]
There haven't been any updates, so I'm closing this to clean up the PR queue and moving to Future milestone.

**Pull request closed without merging**